api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -512,13 +517,41 @@ public void setDosing(DosingInstructions di) {
 	 */
 	@Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder)) {
-			return false;
-		}
-		if (!(otherOrder instanceof DrugOrder)) {
-			return false;
-		}
-		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
-		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+        if (!super.hasSameOrderableAs(otherOrder)) {
+            return false;
+        }
+        if (!(otherOrder instanceof DrugOrder)) {
+            return false;
+        }
+        DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
+
+        if (isNonCodedDrug() || otherDrugOrder.isNonCodedDrug()) {
```

Comment:
Agreed to your point. But it will be caught in validator.
Lets say, we changed to use AND instead of OR. Now below scenario would fail..

Drug Orders with drugs null and one drug non coded value is null.. 
Since the if condition fails, it executes the drug check where it returns true and in reality they both are different orders.